Basic/Pod/MATLAB.pod

```diff
@@ -853,6 +845,101 @@ and open source. It doesn't have PDL's unique features, but it is
 very similar to MATLAB. Scilab comes with B<Xcos> (previously Scicos),
 a graphical system modeler and simulator similar to Simulink.
 
+=head1 COMPARISON: FLOYD-WARSHALL ALGORITHM
+
+In L<graph theory|https://en.wikipedia.org/wiki/Graph_theory>, an
+apparently-simple but difficult problem is the "shortest path" problem,
+of finding the shortest path between any two nodes. A famous solution
+to this, albeit expensive (it is C<O(V^3)> where C<V> is the number of
+vertices) is the Floyd-Warshall algorithm, which iterates through all
+the possible paths.
+
+Both the MATLAB solution and the PDL solution use vectorisation, so
+hopefully this is a useful comparison. The MATLAB version started with
+the code in
+L<code by Giorgos Dim|https://uk.mathworks.com/matlabcentral/fileexchange/67503-floyd-warshall-vectorized>,
+but modified as that code produces an incorrect path matrix.
+
+Sample data (reflected on both the Wikipedia page, and the Rosetta Code
+website) for the weighted-edges matrix is, in PDL format:
+
+  my $we = pdl q[
+   [Inf Inf  -2 Inf]
+   [  4 Inf   3 Inf]
+   [Inf Inf Inf   2]
+   [Inf  -1 Inf Inf]
+  ];
+
+and in MATLAB format:
+
+  A = [0 1/0 -2 1/0; 4 0 3 1/0; 1/0 1/0 0 2; 1/0 -1 1/0 0]
```

Comment:
You can use `Inf` in MATLAB too: `s,\b1/0\b,Inf,g`

Reply:
Great, thanks! Fixed.